### PR TITLE
DMP-1343 Migrate media_request schema to align with modernised_darts_schema

### PR DESCRIPTION
**Pull request by mmmalik1**
### JIRA link (if applicable) ###
(https://tools.hmcts.net/jira/browse/DMP-1343)

### Change description ###

Migration script created:  db/migration/common/V1_236_add_media_request_fields.sql

In order to add:

- media_request.current_owner and setting default value
- media_request_created_by_fk
- media_request_requestor_fk
- media_request_modified_by_fk
- media_request_current_owner_fk

Also, Updated src/main/java/uk/gov/hmcts/darts/audio/entity/MediaRequestEntity.java to add the new field and the relevant @ManyToOne and @JoinColumn annotations.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+ALTER TABLE media_request ADD COLUMN current_owner INTEGER;
+UPDATE media_request SET current_owner=requestor;
+ALTER TABLE media_request ALTER COLUMN current_owner SET NOT NULL;
+
+ALTER TABLE media_request
```
Comment (richardbull) [inline src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql:5]: I'm guessing one of these created_by should be for current_owner?

```
ALTER TABLE media_request   
ADD CONSTRAINT media_request_current_owner_fk
FOREIGN KEY (current_owner) REFERENCES user_account(usr_id);
```

**Review by cmagowan-hmcts — DISMISSED** [review]

File: src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql
[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql:1]: Yes, I have bumped mine up

File: src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql (hunk 1 above)
Comment (richardbull) [reply src/main/resources/db/migration/common/V1_236__Update_media_request_table.sql]: Still

File: src/main/java/uk/gov/hmcts/darts/audio/entity/MediaRequestEntity.java [hunk 2]
```diff
@@ -57,6 +58,10 @@ public class MediaRequestEntity extends CreatedModifiedBaseEntity {
     @JoinColumn(name = HEARING_ID_COLUMN_NAME, nullable = false)
     private HearingEntity hearing;
 
+    @ManyToOne(optional = false)
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/audio/entity/MediaRequestEntity.java:61]: I'm not entirely sure why the other @ManyToOne relationships here don't have LAZY fetch - but thought we could do that here
`@ManyToOne(fetch = FetchType.LAZY, optional = false)`

File: src/main/resources/db/migration/common/V1_241__Update_media_request_table.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ADD CONSTRAINT media_request_requestor_fk
+FOREIGN KEY (requestor) REFERENCES user_account(usr_id);
+
+ALTER TABLE media
+ADD CONSTRAINT media_created_by_fk
+FOREIGN KEY (created_by) REFERENCES user_account(usr_id);
+
+ALTER TABLE media
+ADD CONSTRAINT media_modified_by_fk
+FOREIGN KEY (last_modified_by) REFERENCES user_account(usr_id);
```
Comment (chrisbellingham-hmcts) [inline src/main/resources/db/migration/common/V1_241__Update_media_request_table.sql:19]: Changes to `media` table are out of scope per DMP-1343

File: src/main/java/uk/gov/hmcts/darts/audio/entity/MediaRequestEntity.java (hunk 2 above)
Comment (chrisbellingham-hmcts) [reply src/main/java/uk/gov/hmcts/darts/audio/entity/MediaRequestEntity.java]: Agree, updated.

**Review by chrisdavecm — APPROVED** [review]

**Pull request merged**